array-reduce/main.js

```diff
@@ -0,0 +1,43 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+const sum = numbers.reduce((previousValue, currentValue) => {
+  return previousValue + currentValue;
+});
+console.log('sum: ', sum);
+
+const product = numbers.reduce((previousValue, currentValue) => {
+  return previousValue * currentValue;
+});
+console.log('product: ', product);
+
+const balance = account.filter(item =>
+  item.type === 'deposit').reduce((previousValue, currentValue) => {
+  return previousValue + currentValue.amount;
+}, 0) - account.filter(item =>
+  item.type === 'withdrawal').reduce((previousValue, currentValue) => {
+  return previousValue + currentValue.amount;
+}, 0);
```

Comment:
Try to accomplish this with one `.reduce` and no `.filter`s.

Reply:
<img width="854" alt="Screen Shot 2022-08-26 at 3 47 06 PM" src="https://user-images.githubusercontent.com/106217355/187000022-7679d531-7b6a-43e1-8af8-027a948748ba.png">
<img width="567" alt="Screen Shot 2022-08-26 at 3 47 19 PM" src="https://user-images.githubusercontent.com/106217355/187000038-5020ada0-4df8-4c4b-8aaa-6e3175c8276a.png">